doma-core/src/main/java/org/seasar/doma/jdbc/query/AutoBatchInsertQuery.java

```diff
@@ -167,8 +183,21 @@ protected void prepareSql() {
     }
     builder.cutBackSql(2);
     builder.appendSql(")");
-    PreparedSql sql = builder.build(this::comment);
-    sqls.add(sql);
+  }
+
+  private void assembleUpsertSql(PreparedSqlBuilder builder, Naming naming, Dialect dialect) {
+    UpsertAssemblerContext context =
+        UpsertAssemblerContextBuilder.fromEntity(
+            builder,
+            entityType,
+            duplicateKeyType,
+            naming,
+            dialect,
+            idPropertyTypes,
+            targetPropertyTypes,
+            currentEntity);
+    UpsertAssembler upsertAssemblerQuery = dialect.getUpsertAssembler(context);
```

Comment:
`upsertAssembler` is a more desirable variable name than `upsertAssemblerQuery`.